src/cider/nrepl/middleware/track_state.clj

```diff
@@ -66,13 +74,28 @@
        ;; For some reason, cljs (or piggieback) adds a :test key to the
        ;; var metadata stored in the namespace.
        :interns (merge (misc/update-vals #(dissoc (um/relevant-meta (cljs-meta-with-fn %)) :test) defs)
-                       ;; FIXME: `uses` and `use-macros` are maps from
-                       ;; symbols to namespace names:
-                       ;;     {log reagent.debug, dbg reagent.debug}
-                       ;; Since we don't know the metadata for these
-                       ;; vars, we resign to "guessing" them.
-                       (misc/update-vals (constantly {:arglists '([])}) uses)
-                       (misc/update-vals (constantly {:arglists '([]) :macro true}) use-macros))})
+                       (misc/update-vals um/relevant-meta
+                                         (into {}
+                                               (map (fn [[var-name var-ns]]
+                                                      (let [defs (some->> all-objects
+                                                                          (filter (fn [x]
+                                                                                    (and (map? x)
+                                                                                         (= (:name x)
+                                                                                            var-ns))))
+                                                                          first
+                                                                          :defs)]
+                                                        (if-let [var-meta (some-> defs (get var-name) (get :meta))]
+                                                          [var-name (remove-redundant-quote var-meta)]
+                                                          [var-name {:arglists '([])}]))))
+                                               uses))
```

Comment:
Might be good to leave this in as comments. I also wonder if some function should be extracted out of this, so that the `update-vals` call reads better.